장비 정보 취득 기능

miho030 opened the pull request.

miho030 committed:
환경 정보 획득 기능 구현

- OS 이름, 커널 버전
- 장비 호스트명
- 장비 IP 주소
- 장비 MAC 주소
- 장비 CPU 아키텍처
- 장비 Shell 제어

miho030 committed:
코드 재구성 이전 기능 구현 파일

- 장비 기본 정보 획득(OS,IP,MAC 등)
- 네트워크 어댑터 정보 획득
- 장비 내 활성화된 서비스 목록 획득
- 서비스 목록 중 SSH/FTP 서비스 활성상태 확인
- SSH/FTP 서비스 환경설정/로그 파일 유무 및 경로 확인

npclown committed:
환경 정보 취득 기능 코드 보완(#15)

- 네트워크 어댑터 정보 획득
  - pcap 라이브러리 연결 오류 해결
- 장비 서비스 목록 정보 획득
  - 서비스 구조체 생성 후 관리
- class 코드 보완

miho030 committed:
대응 정책 로더 프로토타입 코드 작성 완료 (#21)

- 대응 정책 파일명 확인 및 파일 경로 확인
- 확인된 파일에 대해 최초생성,수정,최근실행 날짜 조회
- 조사한 정보를 토대로 policyList 구조체 형식을 따르는 ST_POLICY_INFO Vector에 값을 대입하여 저장하는 기능 코드 작성

npclown commented:
@miho030 Vmware에서 라즈베리 운영체제를 올렸을땐 Model과 Serial이 확인이 안되던데, 라즈베리 자체에서만 존재하는 것인지 확인 필요

#22

miho030 committed:
*Cpolicy 기능 구현
- 대응 정책 파일 실행 여부 파악하는 기능 구현 완료
- 대응 정책 파일의 정보(최초 파일생성, 파일수정, 최근 파일실행 날짜)를 조회하는 기능 구현
- 대응 정책 파일의 이름, 관리장비에서의 물리적인 위치(디렉터리 경로)를 파악하는 기능 구현

* CMessage로부터 policy_id, policy_name 값을 받아와 해당 파일을 실행하는 기능 구현(CPolicy::active)
- 대응 정책 "파일"로 관리장비에서 제어하는 경우에 사용 가능.
- 현재 논의된 대응 정책의 종류(sh, c++)에 맞게 실행하는 기능을 우선적으로 작성하였음.
- 테스트는 더미 값으로 실행하였으며, 정상적으로 잘 작동하는 것을 확인하였음.
- 추후 대응 정책의 실행 종류에 따라 Agent의 activate 코드가 보완될 예정임.

miho030 committed:
CPolicy::active 내 반환값 오타 수정

else 부분의 반환값(ActiveRes)의 값을 0이 아닌 1을 가지도록 수정하였음.

miho030 committed:
*CPolicy 내용 수정
- activeFull 함수 코드 작성 완료
- activeFull 함수 테스트 완료. (6개 파일에 대해 실행 테스트)

miho030 committed:
Merge branch 'Added/device' of https://github.com/BoB10-SEDR/Agent into Added/device

miho030 committed:
*Cpolicy 코드 보완
- Cpolicy::active 오타 수정
- Cpolicy::activeFull 기능 구현

npclown committed:
Merge branch 'develop' into Added/device
